runtime/commonMain/src/kotlinx/benchmark/BenchmarkReportFormatter.kt

```diff
@@ -0,0 +1,201 @@
+        val errorLength = columnLength("Error") { it.error.format(3, useGrouping = false) } + padding - 1
+        val unitsLength = columnLength("Units") { unitText(it.config.mode, it.config.outputTimeUnit) } + padding
+
+        return buildString {
+            appendPaddedAfter("Benchmark", nameLength)
+            paramNames.forEach {
+                appendPaddedBefore("($it)", paramLengths[it]!!)
+            }
+            appendPaddedBefore("Mode", modeLength)
+            appendPaddedBefore("Cnt", samplesLength)
+            appendPaddedBefore("Score", scopeLength)
+            append("  ")
+            appendPaddedBefore("Error", errorLength)
+            appendPaddedBefore("Units", unitsLength)
+            appendLine()
+
+            results.forEach { result ->
+                appendPaddedAfter(shortNames[result.benchmark.name]!!, nameLength)
+                paramNames.forEach {
+                    appendPaddedBefore(result.params[it] ?: "N/A", paramLengths[it]!!)
+                }
+                appendPaddedBefore(result.config.mode.toText(), modeLength)
+                appendPaddedBefore(result.values.size.takeIf { it > 1 }?.toString() ?: " ", samplesLength)
+                appendPaddedBefore(result.score.format(3, useGrouping = false), scopeLength)
+                if (result.error.isNaNOrZero()) {
+                    append("  ")
+                    appendPaddedBefore("", errorLength)
+                } else {
+                    append(" \u00B1")
+                    appendPaddedBefore(result.error.format(3, useGrouping = false), errorLength)
+                }
+                appendPaddedBefore(unitText(result.config.mode, result.config.outputTimeUnit), unitsLength)
+                appendLine()
+            }
+        }
+    }
+
+    private fun StringBuilder.appendSpace(l: Int): StringBuilder = append(" ".repeat(l))
+
+    private fun StringBuilder.appendPaddedBefore(value: String, l: Int): StringBuilder =
+        appendSpace(l - value.length).append(value)
+
+    private fun StringBuilder.appendPaddedAfter(value: String, l: Int): StringBuilder =
+        append(value).appendSpace(l - value.length)
+
+
+    /**
+     * Algorithm:
+     *  1. remove package names, if it is the same for all benchmarks
+     *  2. if not, shorthand same package names
+     *
+     *  (jmh similar logic)
+     */
+    private fun denseBenchmarkNames(src: List<String>): Map<String, String> {
+        if (src.isEmpty()) return emptyMap()
+
+        var first = true
+        var prefixCut = false
+
+        val prefix = src.fold(emptyList<String>()) { prefix, s ->
+            val names = s.split(".")
+            if (first) {
+                first = false
+                names.takeWhile { it.toLowerCase() == it }
+            } else {
+                val common = prefix.zip(names).takeWhile { (p, n) -> p == n && n.toLowerCase() == n }
```

Comment:
Minor: `n.toLowerCase() == n` is redundant. If `prefix` already meets `p.toLowerCase() == p`, and `p == n`, then `n.toLowerCase() == n` is true